src/sentry/static/sentry/app/views/settings/components/forms/datePickerField.tsx

```diff
@@ -0,0 +1,106 @@
+import 'react-date-range/dist/styles.css';
+import 'react-date-range/dist/theme/default.css';
+
+import {Calendar} from 'react-date-range';
+import React from 'react';
+import moment from 'moment';
+import styled from '@emotion/styled';
+
+import {inputStyles} from 'app/styles/input';
+import DropdownMenu from 'app/components/dropdownMenu';
+import {IconCalendar} from 'app/icons';
+import space from 'app/styles/space';
+
+import InputField from './inputField';
+
+type Props = Omit<InputField['props'], 'field'>;
+
+function handleChangeDate(onChange, onBlur, date, close) {
+  onChange(date, {});
+  onBlur(date, {});
+
+  // close dropdown menu
+  close();
+}
+
+export default function DatePickerField(props: Props) {
+  return (
+    <InputField
+      {...props}
+      field={({onChange, onBlur, value, disabled, id}) => {
+        const dateObj = new Date(value);
+        const inputValue = !isNaN(dateObj.getTime()) ? dateObj : new Date();
+        const dateString = moment(inputValue).format('LL');
+
+        return (
+          <DropdownMenu keepMenuOpen>
+            {({isOpen, getRootProps, getActorProps, getMenuProps, actions}) => (
+              <DatePickerWrapper {...getRootProps()}>
+                <InputWrapper id={id} {...getActorProps()} isOpen={isOpen}>
+                  <StyledInput readOnly value={dateString} />
+                  <CalendarIcon>
+                    <IconCalendar />
+                  </CalendarIcon>
+                </InputWrapper>
+
+                {isOpen && (
+                  <CalendarMenu {...getMenuProps()}>
+                    <Calendar
+                      disabled={disabled}
+                      date={inputValue}
+                      onChange={date =>
+                        handleChangeDate(onChange, onBlur, date, actions.close)
+                      }
+                    />
+                  </CalendarMenu>
+                )}
+              </DatePickerWrapper>
+            )}
+          </DropdownMenu>
+        );
+      }}
+    />
+  );
+}
+
+type InputWrapperProps = {
+  isOpen: boolean;
+};
+
+const DatePickerWrapper = styled('div')``;
```

Comment:
Maybe this does not need to be a styled component, seeing it's not being utilized anywhere.

Reply:
It's being used here https://github.com/getsentry/sentry/pull/19723/files/d402a17276544e09c78af84738cf6b9b5a6fc876#diff-614709580b8e1d4f1b7f57c8436b10b9R38? It's being used there to semantically name the wrapping element?

Reply:
In most of the cases, names are very helpful, but in this case I'd use only the div instead..does the "DropdownMenu" component only accepts one react element as a child? wouldn't it be better to use <React.Fragment />?

Reply:
Sure, no problem, converted it to a `div` 👍.

As for your `DropdownMenu` suggestion, can you clarify a bit? Do you mean in lieu of a `div` for `DatePickerWrapper`, using `Fragment`? But `Fragment` doesn't accept attributes as it's not a dom node? Am I missing something?

Reply:
No no... I saw now that you are passing "`getRootProps`" props ...so it is fine 👌